internal/blockchain/validate.go

```diff
@@ -1062,7 +1238,7 @@ func (b *BlockChain) checkBlockHeaderPositional(header *wire.BlockHeader, prevNo
 		// diff activation height has been reached.
 		blockHeight := prevNode.height + 1
 		if b.minTestNetTarget != nil &&
-			expDiff <= standalone.BigToCompact(b.minTestNetTarget) &&
+			header.Bits <= standalone.BigToCompact(b.minTestNetTarget) &&
```

Comment:
i believe this could use b.minTestNetDiffBits instead of doing the BigToCompact calculation again.